draco/analysis/powerspec.py

```diff
@@ -0,0 +1,1568 @@
+            "blackman_nuttall",
+            "blackman_harris",
+            "tukey-0.5",
+        ],
+        default="tukey-0.5",
+    )
+    ew_min = config.Property(proptype=float, default=9.0)
+    ew_max = config.Property(proptype=float, default=50.0)
+    ns_bl = config.Property(proptype=float, default=60.0)
+
+    def process(self, ds):
+        """Estimate the spatial transform of the delay map.
+
+        Parameters
+        ----------
+        ds : containers.DelayTransform
+          The delay map, whose spatial transform will be estimated.
+
+        Returns
+        -------
+        spatial cube : containers.SpatialDelayCube
+           The data cube in (delay,u,v) domain.
+        """
+        if not isinstance(ds, containers.DelayTransform):
+            raise ValueError(
+                f"Input container must be instance of DelayTransform (received {ds.__class__})"
+            )
+
+        ds.redistribute("delay")
+
+        # Extract required data axes
+        delay = ds.index_map["delay"]  # micro-sec
+        el = ds.index_map["el"]
+        pol = ds.index_map["pol"]
+        ra = ds.index_map["sample"]  # deg
+        dec = sza2dec(el)  # deg
+        freq = ds.attrs["freq"]  # MHz
+        wl = C / (freq * 1e6)  # wavelength in meter
+
+        # Unpack the baseline axis of the delay spectrum
+        # and reshape it as (pol,delay,ra,el)
+        axes = list(ds.attrs["baseline_axes"])
+        shp = tuple([ds.index_map[ax].size for ax in axes])
+        data_view = ds.spectrum[:].local_array.reshape(*shp, ra.size, -1)[0, :]
+        data_view = np.swapaxes(data_view, 1, 3)
+        data_view = mpiarray.MPIArray.wrap(data_view, axis=1, comm=ds.comm)
+        # redistribute over delay axis
+        data_view = data_view.redistribute(axis=1)
+
+        # Estimate the Fourier modes
+        nu_c = freq[int(freq.size / 2.0)]  # central freq of the band
+        redshift = f21 / nu_c - 1  # redshift at the center of the band
+        kx, ky, u, v, k_parallel = get_fourier_modes(ra, dec, delay * 1e-6, redshift)
+
+        # Estimate the uv-mask
+        uv_mask = spatial_mask(
+            kx, ky, self.ew_min, self.ew_max, self.ns_bl, wl.min(), wl.max(), redshift
+        )
+
+        # Estimate the volume of the cube
+        vol_cube = vol_normalization(ra, dec, freq, redshift)
+
+        # Initialise the spatial transformed data cube container
+        vis_cube = containers.SpatialDelayCube(
+            pol=pol, delay=delay, kx=kx, ky=ky, attrs_from=ds
+        )
+        vis_cube.redistribute("delay")
+        vis_cube.data_tau_uv[:] = 0.0
+
+        # Save the central freq of the band and the
+        # corresponding redshift and the cube volume in the attrs
+        vis_cube.attrs["freq_center"] = nu_c
+        vis_cube.attrs["redshift"] = redshift
+        vis_cube.attrs["volume"] = vol_cube
+
+        # Create index map for (u,v) coordinates and k_parallel
+        fourier_axes = ["u", "v", "k_parallel", "uv_mask"]
```

Comment:
I mention this in the `containers` file as well, but I think that these should actually be datasets in the container rather than index maps, since two containers with the same `kx` `ky` axes could have different `u` and `v` depending on the cosmology used (I think).

It's also cleaner since `index_map` entries are generally expected to be used by at least one dataset, so having some extra axes doesn't fit into the container code design as well.